daemon/container_unix.go

```diff
@@ -1455,6 +1455,74 @@ func (container *Container) ipcMounts() []execdriver.Mount {
 	return mounts
 }
 
+func updateCommand(c *execdriver.Command, resources runconfig.Resources) {
+	c.Resources.BlkioWeight = resources.BlkioWeight
+	c.Resources.CPUShares = resources.CPUShares
+	c.Resources.CPUPeriod = resources.CPUPeriod
+	c.Resources.CPUQuota = resources.CPUQuota
+	c.Resources.CpusetCpus = resources.CpusetCpus
+	c.Resources.CpusetMems = resources.CpusetMems
+	c.Resources.Memory = resources.Memory
+	c.Resources.MemorySwap = resources.MemorySwap
+	c.Resources.MemoryReservation = resources.MemoryReservation
+	c.Resources.KernelMemory = resources.KernelMemory
+}
+
+func (container *Container) setContainer(hostConfig *runconfig.HostConfig) error {
+	container.Lock()
+
+	resources := hostConfig.Resources
+	cResources := &container.hostConfig.Resources
+	if resources.BlkioWeight != 0 {
+		cResources.BlkioWeight = resources.BlkioWeight
+	}
+	if resources.CPUShares != 0 {
+		cResources.CPUShares = resources.CPUShares
+	}
+	if resources.CPUPeriod != 0 {
+		cResources.CPUPeriod = resources.CPUPeriod
+	}
+	if resources.CPUQuota != 0 {
```

Comment:
yep that would work, but I think the default golang json parser is going to get in the way because I don't think, by default, it tells you if a particular property is in the json or not - all you see is zero in the struct afterwards. I think it might require a more specialized parser and then we'd need to make sure we only include "changed" fields in the json, not the entire struct.  Either that, or do a GET to get to current values and then we can pass them all, but the extra message exchange and risk of one of those "current values" changes on you is a problem.  I'd prefer a delta-update.